### ISSUE TITLE
Make regular expression capture groups available in the command that matches the path

### ISSUE THREAD

**Issue by rmwiseman** [aler9/rtsp-simple-server#642]
Discussed in https://github.com/aler9/rtsp-simple-server/discussions/619

(NOTE: I did add this using the enhancement template, but it seems to have lost its label.)

If I use brackets in a regular expression path, I would like those capture groups to be available for use in the command (`runOnDemand` / `runOnPublish` / `runOnRead`) or `source` relating to that regular expression path.

For example:

```
paths:
  "~^small/([A-Za-z-]{36})$":
    source: rtsp://small-frames/cameras/$1
```

or

```
paths:
  "~^([0-9]{2,4})h/([A-Za-z-]{36})$":
    runOnDemand: ffmpeg -hide_banner -re -i "rtsp://user:pass@source-video/$2" -vcodec libx264 -filter "scale=w=-1:h=$1" -f rtsp rtsp://127.0.0.1:$RTSP_PORT/$RTSP_PATH
```

If `$1` and `$2` etc. aren't possible then `$M1` and `$M2` or similar could be used.

**Comment by dorinclisu** [comment]
I was going to ask the same.

**Comment by aler9** [comment]
This has been implemented in main and will be added to the next release.

**Comment by dorinclisu** [comment]
Thanks, I had a look at the result and I realized my use case is a bit different.
Basically what I need is this, which cannot work with subprocess environment variables:
```
paths:
  "~^([0-9]{2,4})h/([A-Za-z-]{36})$":
    source: redirect
    sourceRedirect: rtsp://other-server/{{g1}}-{{g2}}
 ```

**Comment by aler9** [comment]
added in v0.17.12

**Comment by rmwiseman** [comment]
Thanks for implementing this.  I'm only just getting around to trying it out, but for some reason I can't make it work!  It seems that, for me, the `$1` variable isn't set in the `runOnDemand` command, so it fails.   My path is e.g. `"~^h([0-9]*)/7b50acef-bb8c-410a-8ccd-bf46e5dc6a06"` (I included the double quotes in the YAML file as the examples I've seen do that too), the idea being that if I want a video stream that's, say, 200px high, I use the path `h200/7b50acef-bb8c-410a-8ccd-bf46e5dc6a06`.  A simplified version of my `runOnDemand` command is: `ffmpeg -i rtsp://127.0.0.1:8554/full/7b50acef-bb8c-410a-8ccd-bf46e5dc6a06 -vcodec libx264 -filter scale=w=-1:h=$1 -f rtsp rtsp://127.0.0.1:$RTSP_PORT/$RTSP_PATH` and from the output I see, it looks like `$1` is unset.  Am I doing something stupid?  (I couldn't find any examples in the documentation, so I had to guess what to do from looking at the code.)

**Comment by aler9** [comment]
Hello, it turned out that variables like $1 and $2 are not available under Linux since they're reserved for command line arguments.
I renamed the variables into $G1 and $G2, in this way they won't be overridden.
This change will be available from the next release.

**Comment by rmwiseman** [comment]
I did wonder if that might be the case.  Thanks for the fix! :-)

**Comment by aler9** [comment]
The fix has been added in v0.17.15

**Comment by rmwiseman** [comment]
Many thanks for the fix.

I don't know if it's regexp-related, but one weird issue I have is that sometimes (connecting using VLC) I have to try multiple times (sometimes lots!) for it to work.  I don't have that problem with non-regexp paths, at least.  The log just keeps saying nothing is publishing to the path, but clearly eventually something does.  I initially thought that the problem was that I was using `*`s in my regexp, since ones with `+`s seemed to work all the time, but I've found that if I retry enough times, it usually eventually works.
